packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx

```diff
@@ -0,0 +1,374 @@
+            )
+            if (!kvCache) return
+            const result: BindingProof[] = walletsList.reduce<BindingProof[]>((res, x) => {
+                x.isDefault = 0
+                x.isPublic = 1
+                const temp = (kvCache?.content[PluginId.Tip] as BindingProof[]).filter((i) =>
+                    isSameAddress(x.identity, i.identity),
+                )
+                if (temp && temp.length > 0) {
+                    x.isDefault = temp[0].isDefault
+                    x.isPublic = temp[0].isPublic
+                }
+                res.push(x)
+                return res
+            }, [])
+            const idx = result.findIndex((i) => i.isDefault)
+            if (idx !== -1) {
+                result.unshift(result.splice(idx, 1)[0])
+            } else {
+                result[0].isDefault = 1
+            }
+
+            setRawWalletList(result)
+            setRawPatchData(result)
+            return
+        }
+        walletsList.forEach((x, idx) => {
+            x.isPublic = 1
+            x.isDefault = 0
+            if (idx === 0) {
+                x.isDefault = 1
+                return
+            }
+        })
+        setRawWalletList(cloneDeep(walletsList))
+        setRawPatchData(cloneDeep(walletsList))
+    }, [proofRes, kv, bodyView])
+
+    const onCancel = () => {
+        setRawPatchData(cloneDeep(rawWalletList))
+        setHasChanged(false)
+    }
+
+    const refresh = () => {
+        setBodyView(BodyViewSteps.main)
+        retryProof()
+        retryKv()
+    }
+    const WalletButton = () => {
+        const { classes } = useStyles()
+        return (
+            (bodyView !== BodyViewSteps.addWallet && (
+                <div className={classes.btnContainer}>
+                    <Button
+                        onClick={() => {
+                            if (bodyView === BodyViewSteps.wallets) {
+                                setBodyView(BodyViewSteps.addWallet)
+                            } else {
+                                setBodyView(BodyViewSteps.wallets)
+                            }
+                        }}
+                        className={classes.walletBtn}
+                        variant="contained"
+                        size="small">
+                        {bodyView === BodyViewSteps.wallets ? BodyViewSteps.addWallet : BodyViewSteps.wallets}
+                    </Button>
+                </div>
+            )) ||
+            null
+        )
+    }
+    const setAsDefault = (idx: number) => {
+        const changed = cloneDeep(rawPatchData)
+        changed.forEach((x: any) => (x.isDefault = 0))
+        changed[idx].isDefault = 1
+        const defaultItem = changed[idx]
+        changed.splice(idx, 1)
+        changed.sort(
+            (a: BindingProof, b: BindingProof) =>
+                Number.parseInt(b.last_checked_at, 10) - Number.parseInt(a.last_checked_at, 10),
+        )
```

Comment:
`changed` is `BindingProof[]`, then typings can be omit here.
```suggestion
        changed.forEach((x) => (x.isDefault = 0))
        changed[idx].isDefault = 1
        const defaultItem = changed[idx]
        changed.splice(idx, 1)
        changed.sort((a, b) => Number.parseInt(b.last_checked_at, 10) - Number.parseInt(a.last_checked_at, 10))
```